File: src/main/java/net/riches/common_utils/internal/Pair.java
package net.riches.common_utils.internal;

public class Pair<F, S> {
    private final F first;
    private final S second;

    public Pair(F first, S second) {
        this.first = first;
        this.second = second;
    }

    public F getFirst() {
        return first;
    }

    public S getSecond() {
        return second;
    }

    public static <T,R> Pair<T,R> pair(T first, R second) {
        return new Pair<>(first, second);
    }
}

File: src/main/java/net/riches/common_utils/internal/Pair.kt
package net.riches.common_utils.internal

class Pair<F, S>(first: F, second: S) {
    private var f: F? = first
    private var s: S? = second

    fun getFirst(): F? {
        return f
    }

    fun getSecond(): S? {
        return s
    }

    fun <T, R> pair(first: F, second: S): Pair<F, S> {
        return Pair(first, second)
    }
}